proto/opamp.proto

```diff
@@ -609,10 +609,24 @@ message AgentDescription {
     //   with this Agent.
     repeated KeyValue non_identifying_attributes = 2;
 
+    // Details about the components available in the agent.
+    ComponentDetails component_details = 3;
```

Comment:
This mirrors ComponentHealth (single root component), but I'm not sure if there's value in having a "root" ComponentDetails. The root component would just be the agent itself, and I'd expect metadata about the root component to be in identifying/non-identifying attributes.